src/brsTypes/Callable.ts

```diff
@@ -255,6 +257,24 @@ export class Callable implements Brs.BrsValue {
                 return;
             }
 
+            if (
+                expected.type.kind === Brs.ValueKind.Float &&
+                received.kind === Brs.ValueKind.Int32
+            ) {
+                let old = args[index] as Int32;
```

Comment:
Consider using the `received` variable in place of this new `old` one — the TypeScript compiler already knows it's an Int32 because of the `ValueKind` comparison on line 262, so there's no need to re-cast it here!